File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Comerciales/CatalogosComercialesForm.ts
import { LookupEditor, IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CatalogosComercialesForm {
    IdtipoCatalogo: LookupEditor;
    IdClave: IntegerEditor;
    Descripcion: StringEditor;
    Activo: BooleanEditor;
}

export class CatalogosComercialesForm extends PrefixedContext {
    static formKey = 'Comerciales.CatalogosComerciales';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatalogosComercialesForm.init)  {
            CatalogosComercialesForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;
            var w3 = BooleanEditor;

            initFormType(CatalogosComercialesForm, [
                'IdtipoCatalogo', w0,
                'IdClave', w1,
                'Descripcion', w2,
                'Activo', w3
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/Comerciales/CatalogosComerciales/CatalogosComercialesForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.Comerciales.Forms;

[FormScript("Comerciales.CatalogosComerciales")]
[BasedOnRow(typeof(CatalogosComercialesRow), CheckNames = true)]
public class CatalogosComercialesForm
{
    [Editable(false)]
    public int IdtipoCatalogo { get; set; }
    [Editable(false)]
    public int IdClave { get; set; }
    public string Descripcion { get; set; }
    [BooleanEditor]
    public int Activo { get; set; }
    //public DateTime DtRegistro { get; set; }
}